deps/rabbitmq_prometheus/test/rabbit_prometheus_http_SUITE.erl

```diff
@@ -84,6 +96,72 @@ init_per_group(per_object_endpoint_metrics, Config0) ->
     ]},
     Config1 = rabbit_ct_helpers:merge_app_env(Config0, PathConfig),
     init_per_group(aggregated_metrics, Config1);
+init_per_group(core_metrics, Config0) ->
+    FilterEnv = {rabbitmq_prometheus, [{core_metrics_default_vhosts, [<<"vhost-2">>]},
+                                       {core_metrics_default_families, [queue_coarse_metrics]}]},
+
+    Config1 = init_per_group(core_metrics, rabbit_ct_helpers:merge_app_env(Config0, FilterEnv), []),
+
+    rabbit_ct_broker_helpers:add_vhost(Config1, 0, <<"vhost-1">>, <<"guest">>),
+    rabbit_ct_broker_helpers:set_full_permissions(Config1, <<"vhost-1">>),
+    VHost1Conn = rabbit_ct_client_helpers:open_unmanaged_connection(Config1, 0, <<"vhost-1">>),
+    {ok, VHost1Ch} = amqp_connection:open_channel(VHost1Conn),
+
+    rabbit_ct_broker_helpers:add_vhost(Config1, 0, <<"vhost-2">>, <<"guest">>),
+    rabbit_ct_broker_helpers:set_full_permissions(Config1, <<"vhost-2">>),
+    VHost2Conn = rabbit_ct_client_helpers:open_unmanaged_connection(Config1, 0, <<"vhost-2">>),
+    {ok, VHost2Ch} = amqp_connection:open_channel(VHost2Conn),
+
+    DefaultCh = rabbit_ct_client_helpers:open_channel(Config1),
+
+    [
+     (fun () ->
+              QPart = case VHost of
+                          <<"/">> -> <<"default">>;
+                          _ -> VHost
+                      end,
+              QName = << QPart/binary, "-", Q/binary>>,
+              #'queue.declare_ok'{} = amqp_channel:call(Ch,
+                                                        #'queue.declare'{queue = QName,
+                                                                         durable = true
+
+                                                                        }),
+              lists:foreach( fun (_) ->
+                                     amqp_channel:cast(Ch,
+                                                       #'basic.publish'{routing_key = QName},
+                                                       #amqp_msg{payload = <<"msg">>})
+                             end, lists:seq(1, MsgNum) )
+      end)()
+     || {VHost, Ch, MsgNum} <- [{<<"/">>, DefaultCh, 3}, {<<"vhost-1">>, VHost1Ch, 7}, {<<"vhost-2">>, VHost2Ch, 11}],
+        Q <- [ <<"queue-with-messages">>, <<"queue-with-consumer">> ]
+    ],
+
+    DefaultConsumer = sleeping_consumer(),
+    #'basic.consume_ok'{consumer_tag = DefaultCTag} =
+        amqp_channel:subscribe(DefaultCh, #'basic.consume'{queue = <<"default-queue-with-consumer">>}, DefaultConsumer),
+
+    VHost1Consumer = sleeping_consumer(),
+    #'basic.consume_ok'{consumer_tag = VHost1CTag} =
+        amqp_channel:subscribe(VHost1Ch, #'basic.consume'{queue = <<"vhost-1-queue-with-consumer">>}, VHost1Consumer),
+
+    VHost2Consumer = sleeping_consumer(),
+    #'basic.consume_ok'{consumer_tag = VHost2CTag} =
+        amqp_channel:subscribe(VHost2Ch, #'basic.consume'{queue = <<"vhost-2-queue-with-consumer">>}, VHost2Consumer),
+
+    timer:sleep(10000),
```

Comment:
What underlying condition is this sleep meant to account for?

Reply:
I did it by analogy with the other test cases in this file. It is about `collect_statistics_interval`, and this sleep makes sure that all stats will be properly accounted in ETS tables (where prometheus plugin reads them from).

Reply:
Various other tests reduce that interval when setting up the test (e.g. https://github.com/rabbitmq/rabbitmq-server/blob/3f1b228dccd698e6f7bc07420387064cf2853597/deps/rabbitmq_stream_management/test/http_SUITE.erl#L39-L45). I wonder if it is worth doing the same here so that we don't have to sleep 10s per test.

Reply:
I've reduced this and sleep time to 1 second, works on my machine.